yt_dlp/extractor/bilibili.py

```diff
@@ -938,20 +940,44 @@ def _get_subtitles(self, *, ep_id=None, aid=None):
                 'episode_id': ep_id,
                 'aid': aid,
             })) or {}
-        subtitles = {}
-        for sub in sub_json.get('subtitles') or []:
-            sub_url = sub.get('url')
-            if not sub_url:
-                continue
-            sub_data = self._download_json(
-                sub_url, ep_id or aid, errnote='Unable to download subtitles', fatal=False,
-                note='Downloading subtitles%s' % f' for {sub["lang"]}' if sub.get('lang') else '')
-            if not sub_data:
-                continue
-            subtitles.setdefault(sub.get('lang_key', 'en'), []).append({
-                'ext': 'srt',
-                'data': self.json2srt(sub_data)
-            })
+
+        for sub in traverse_obj(sub_json, ('subtitles', lambda _, v: v['url'])):
+            sub_ext = determine_ext(sub['url'])
+            sub_lang = sub.get('lang_key') or 'en'
+
+            sub_data = self._download_webpage(
+                sub['url'], ep_id or aid, fatal=False,
+                encoding='utf-8-sig' if sub_ext == 'ass' else None,
+                note=f'Downloading subtitles{format_field(sub, "lang", "for %s")} ({sub_lang})',
```

Comment:
Missing a space here
```suggestion
                note=f'Downloading subtitles{format_field(sub, "lang", " for %s")} ({sub_lang})',
```